src/handlers/NameWrapper.ts

```diff
@@ -0,0 +1,209 @@
+import { type Context, type Event, type EventNames } from "ponder:registry";
+import { domains, wrappedDomains } from "ponder:schema";
+import { type Address, type Hex, hexToBytes } from "viem";
+import { bigintMax } from "../lib/helpers";
+import { makeEventId } from "../lib/ids";
+import {
+  ETH_NODE,
+  checkPccBurned,
+  decodeDNSPacketBytes,
+  tokenIdToLabel,
+} from "../lib/subname-helpers";
+import { upsertAccount } from "../lib/upserts";
+
+// if the wrappedDomain in question has pcc burned (?) and a higher (?) expiry date, update the domain's expiryDate
+async function materializeDomainExpiryDate(context: Context, node: Hex) {
+  const wrappedDomain = await context.db.find(wrappedDomains, { id: node });
+  if (!wrappedDomain) throw new Error(`Expected WrappedDomain(${node})`);
+
+  // ignore if pcc not burned
+  if (!checkPccBurned(wrappedDomain.fuses)) return;
+
+  // update the domain's expiry to the greater of the two
+  await context.db.update(domains, { id: node }).set((domain) => ({
+    expiryDate: bigintMax(domain.expiryDate ?? 0n, wrappedDomain.expiryDate),
+  }));
+}
+
+async function handleTransfer(
+  context: Context,
+  event: Event<EventNames>,
```

Comment:
Can we define a specific type for event and it's args? Or is that not possible at the moment?

Reply:
in this case the `Event<EventNames>` type is the union of all possible events, which more or less gives us access to all of the properties of event besides `args`, since those are identical between the generate event types. perhaps this could be typealiased to something like `AnyEvent`, wdyt?

Reply:
I guess we might not need the `event` param on this function at all.

All callers of that function already include `tokenId: Hex` & `to: Hex` params, and the `event.args` value include those params as well.